tests/test_builtins.py

```diff
@@ -453,6 +459,243 @@ def test_setdefault(self):
 
 class TestList(TestPenty):
 
+    def test_add(self):
+        self.assertIsType(
+            'x + y', pentyping.List[{int}],
+            env={'x': pentyping.List[{int}],
+                 'y': pentyping.List[{int}]}
+        )
+        self.assertIsType(
+            'x + y', pentyping.List[{int, float}],
+            env={'x': pentyping.List[{int}],
+                 'y': pentyping.List[{float}]}
+        )
+        with self.assertRaises(TypeError):
+            self.assertIsType('x + y', None, env={'x': pentyping.List[{int}],
+                                                  'y': int})
+
+    def test_bool(self):
+        self.assertIsType('bool(list())', pentyping.Cst[False], env={})
+        self.assertIsType('bool(x)', bool, env={'x': pentyping.List[{int}]})
+
+    def test_contains(self):
+        self.assertIsType(
+            'x in y',
+            bool,
+            env={'x': int, 'y': pentyping.List[{float}]}
+        )
+
+    def test_delitem(self):
+        # TODO 'del x[k]' fails to compile, I need to figure out how to test it
```

Comment:
it fails to compile because it's a statement, and we only check expression here. You can either call the magic method explicitly, or add the test in `test_stmt.py`